gateway-service-admin/src/main/java/org/apache/knox/gateway/service/admin/TopologiesResource.java

```diff
@@ -424,6 +429,14 @@ public Response uploadSimpleDescriptor(@PathParam("name") String name,
     return response;
   }
 
+  private boolean existingGeneratedTopology(String fileName, TopologyService topologyService, GatewayConfig config) {
+    for (org.apache.knox.gateway.topology.Topology topology : topologyService.getTopologies()) {
```

Comment:
I do not see where we check if the given topology is `readOnly`. This implementation will prevent creating/overwriting all topologies, won't it?